### PR TITLE
RHINENG-7061: add intersystems field to system_profile

### PR DESCRIPTION
**Pull request by JoySnow**
[RHINENG-7061](https://issues.redhat.com/browse/RHINENG-7061) 

Inventory-schema PR at https://github.com/RedHatInsights/inventory-schemas/pull/122

### CODE REVIEW

**Comment by JoySnow** [comment]
The local test looks good with:  
```
--------------
Rules Executed
--------------
[META] src.puptoo.process.profile.system_profile
------------------------------------------------

    iris                   : {'iris_running_instances': [{'instance_name': 'IRIS',
                                                          'product': 'IRIS',
                                                          'version': '2023.1'},
                                                         {'instance_name': 'IRIS2',
                                                          'product': 'IRIS',
                                                          'version': '2023.1'}],
                              'iris_system': True}
```

File: src/puptoo/process/profile.py [hunk 1]
```diff
@@ -644,6 +650,42 @@ def system_profile(
     if gcp_network_interfaces:
         profile["public_ipv4_addresses"] = _remove_empty_string(gcp_network_interfaces.public_ips)
 
+    if iris_cpfs and iris_list:
+        try:
+            iris_profile = {}
```
Comment (r14chandra) [inline src/puptoo/process/profile.py:655]: Before merging this PR, we would also be required to add a schema 
under inventory-schema repo [like this](https://github.com/RedHatInsights/inventory-schemas/blob/master/schemas/system_profile/v1.yaml#L413). Please have a look.

File: src/puptoo/process/profile.py [hunk 2]
```diff
@@ -644,6 +650,42 @@ def system_profile(
...
+                        cpf_file_path = os.path.join(conf_directory, conf_file)
+                        if instance_name and cpf_file_path and cpf_file_path in iris_cpf_result:
+                            instance_info = _remove_empties({
+                                "instance_name": instance['instance_name'],
+                                "product": iris_cpf_result[cpf_file_path]["product"],
+                                "version": iris_cpf_result[cpf_file_path]["version"],
+                            })
+                            if instance_info:
+                                iris_profile["iris_running_instances"].append(instance_info)
+            profile["iris"] = _remove_empties(iris_profile)
```
Comment (r14chandra) [inline src/puptoo/process/profile.py:684]: I think we don't require `_remove_empties` here. Ultimately, we do apply this on the whole profile [here](https://github.com/RedHatInsights/insights-puptoo/blob/4700ed6de1e0287c6a4b228356f643d07d5d182c/src/puptoo/process/profile.py#L690). Let me know if this makes sense.

**Review by r14chandra — CHANGES_REQUESTED** [review]
Thanks @JoySnow for working on this!
The PR looks good. I have added two suggestions. Please have a look. Thanks!

File: src/puptoo/process/profile.py (hunk 1 above)
Comment (JoySnow) [reply src/puptoo/process/profile.py:655]: Thanks for the comments @r14chandra . Adding the related schema to inventory-schema repo at https://github.com/RedHatInsights/inventory-schemas/pull/122 .

File: src/puptoo/process/profile.py (hunk 2 above)
Comment (JoySnow) [reply src/puptoo/process/profile.py:684]: Hey @r14chandra , 

Thanks for pointing this out too. 

I noticed that this `_remove_empties()` be called [here](https://github.com/RedHatInsights/insights-puptoo/blob/4700ed6de1e0287c6a4b228356f643d07d5d182c/src/puptoo/process/profile.py#L690) will only remove the empty values in first level of `profile`. 

And In this `profile["iris"]`, it may contains cases like: 
```
{
    "iris": {
        "iris_system": True,
        "iris_running_instances": []
    }
}
```
I am not sure if such empty `profile["iris"]["iris_running_instances"]` is acceptable to inventory service or not. Intent to remove the empty `profile["iris"]["iris_running_instances"]` here.

Any best practice for such corner cases?

Comment (r14chandra) [reply src/puptoo/process/profile.py:684]: In that case, I think we should drop that field ie. `iris` altogether as this won't contain any relevant data. Does it make sense?

Comment (r14chandra) [reply src/puptoo/process/profile.py:684]: > Any best practice for such corner cases?

It is hard to know if such an edge case as above is valid or not but I try a few cases that are possible and try to include some tests so that it would be easier for others to understand what they would be getting. 
Later if we get any issue with any scenario then we can anyway update the code and add a test around it.

File: src/puptoo/process/profile.py [hunk 3]
```diff
@@ -644,6 +650,42 @@ def system_profile(
     if gcp_network_interfaces:
         profile["public_ipv4_addresses"] = _remove_empty_string(gcp_network_interfaces.public_ips)
 
+    if iris_cpfs and iris_list:
+        try:
+            iris_profile = {}
+            iris_profile["iris_system"] = True
```
Comment (r14chandra) [inline src/puptoo/process/profile.py:656]: Do we really need this field - `iris_system` Just represent the system is InterSysytems? IMO, checking if the array i.e. `iris_running_instances` is not empty then also we can conclude the same. Any thoughts?

**Review by r14chandra — CHANGES_REQUESTED** [review]

File: src/puptoo/process/profile.py (hunk 3 above)
Comment (r14chandra) [reply src/puptoo/process/profile.py:656]: Okay, then I think we are good. 
Do we need to change the naming of those fields as per the JIRA card?

Comment (JoySnow) [reply src/puptoo/process/profile.py:656]: Yep, I will update accordingly after confirming the detail changes in [Jira](https://issues.redhat.com/browse/RHINENG-7061) .

**Comment by JoySnow** [comment]
Get the confirmation of the field name changes in [Jira](https://issues.redhat.com/browse/RHINENG-7061) :

> Use "intersystems" instead of "iris" as field name.

Updated the code accordingly: (Amended the commit message to include the "intersystems" word.)

- The local test looks good with:

    ```
        intersystems           : {'is_intersystems': True,
                                  'running_instances': [{'instance_name': 'IRIS',
                                                         'product': 'IRIS',
                                                         'version': '2023.1'},
                                                        {'instance_name': 'IRIS2',
                                                         'product': 'IRIS',
                                                         'version': '2023.1'}]}
    ```

- inventory-schema local test also passed, against the latest code in https://github.com/RedHatInsights/inventory-schemas/pull/122 .

**Comment by JoySnow** [comment]
Updated the code accordingly after the field name changes confirmation in [Jira](https://issues.redhat.com/browse/RHINENG-7061) .
And local test looks fine: details at https://github.com/RedHatInsights/insights-puptoo/pull/316#issuecomment-1907878622 

Hi @r14chandra , could you kindly help review the changes?

**Review by r14chandra — APPROVED** [review]
Thanks @JoySnow for the PR and solving the comments.
Looks good 👍🏻

**Comment by JoySnow** [comment]
Hi @xiangce , could you also have a look at this update?

**Review by xiangce — APPROVED** [review]
LGTM
